### ISSUE TITLE
Rofi wayland git

### ISSUE THREAD

**Issue by nx-smul** [Rolv-Apneseth/rofi-games#7]
getting this error with rofi wayland git fork 

![image](https://github.com/Rolv-Apneseth/rofi-games/assets/68707319/69c48bf6-bca6-40a6-b186-05855fca636f)

**Comment by Rolv-Apneseth** [comment]
Yeah I used to use that one and started getting this error after some update so I just went back to regular rofi. I'll have a look when I get a chance though, might open an issue on their repo and see if they know what's causing it.

**Comment by Rolv-Apneseth** [comment]
Just as a temporary workaround, as mentioned [here](https://github.com/svenstaro/rofi-calc/issues/107) (this plugin also has the same issue) you can just use `rofi-lbonn-wayland` (not the `-git`) version and that should work, even though AUR says that's outdated

**Comment by nx-smul** [comment]
winndow option dosen't work on no git version :""(

**Comment by Rolv-Apneseth** [comment]
Weird, installing `rofi-lbonn-wayland` fixes it for me. According to [here](https://github.com/Mange/rofi-emoji/issues/65) this is probably due to an update to the `rofi` api which is probably already in the `-git` version I would guess. I'll continue looking into how to fix this one.

**Comment by nx-smul** [comment]
I thought I'm the only one. Thank you

**Comment by Rolv-Apneseth** [comment]
Whoops sorry didn't mean to have that close automatically. Let me know if that works for you @nx-smul. Note that to re-install the package properly, you should run something like `paru -S rofi-games --rebuild` so that it doesn't skip the build step, which is essential for setting the correct flag to use the newer version which should be compatible with `rofi-lbonn-wayland-git`

**Comment by zDyanTB** [comment]
Hello. I'm using Fedora and I'm getting the same error, but the parameters are different.
![image](https://github.com/Rolv-Apneseth/rofi-games/assets/102763838/6487386a-bace-4f74-8956-7758e1a072f0)

![image](https://github.com/Rolv-Apneseth/rofi-games/assets/102763838/e37659b6-fd32-433b-8fb8-7be252f5eba9)
Rofi version is on 1.7.5

**Comment by Rolv-Apneseth** [comment]
@zDyanTB how did you install the `rofi-games`? It seems to have built for the next version of `rofi`

**Comment by zDyanTB** [comment]
I've tried with `just`, after, manual. The error persists. I did not use the tag to use the next version.

**Comment by Rolv-Apneseth** [comment]
Oh sorry, the `just` method should have used the correct flag, which for you should be no flag. Can you give me the exact output of `rofi -version`?

**Comment by Rolv-Apneseth** [comment]
Strange, not sure why that's happening. Could you give me the exact output of `rofi -version`?

For the manual install, did you uninstall the previous version before doing that? Just not sure how it's possible to get that error without the extra flag to specifically build it for the next version

**Comment by zDyanTB** [comment]
❯ rofi -version             
Version: 1.7.5+wayland2

[Installation log](https://pastebin.com/kLseypvq)

Uninstall, you mean, remove the folder and the games.so files? I Did

**Comment by Rolv-Apneseth** [comment]
Sorry about this. I can see why the `just` install fails, since I didn't know the output could look like that, but I have no idea why the manual install would fail.

What is the output of `pkg-config --variable pluginsdir rofi`? Perhaps the version installed by `just` still exists in that folder?

**Comment by Rolv-Apneseth** [comment]
If you have some time would you mind letting me know if installing with `just` with that PR works for you? Thanks for reporting this.

**Comment by zDyanTB** [comment]
> Sorry about this. I can see why the `just` install fails, since I didn't know the output could look like that, but I have no idea why the manual install would fail.
> 
> What is the output of `pkg-config --variable pluginsdir rofi`? Perhaps the version installed by `just` still exists in that folder?

```console
❯ pkg-config --variable pluginsdir rofi
/usr/lib64/rofi/
```
<br>
I uninstalled using find and removing the `games.so` files

```console
find / -name games.so 2>/dev/null -exec rm -f {} +
```

**Comment by zDyanTB** [comment]
> If you have some time would you mind letting me know if installing with `just` with that PR works for you? Thanks for reporting this.

wym "PR"?

**Comment by zDyanTB** [comment]
> > If you have some time would you mind letting me know if installing with `just` with that PR works for you? Thanks for reporting this.
> 
> wym "PR"?

Oh I got it.

Here's my terminal:
```console

~ 
❯ rofi -modi games -show games -theme games

(process:25959): Rofi-WARNING **: 21:25:02.075: ABI version of plugin: 'games.so' does not match: 00000007 expecting: 00000006

(process:25959): Rofi-WARNING **: 21:25:02.084: Failed to parse theme: "games"

(process:25959): Rofi-WARNING **: 21:25:02.084: Failed to load theme. Try to load default: 
[1]    25959 terminated  rofi -modi games -show games -theme games

~ 
❯ rofi -modi games -show games -theme games-default

(process:26016): Rofi-WARNING **: 21:25:06.818: ABI version of plugin: 'games.so' does not match: 00000007 expecting: 00000006

~ took 5s 
❯ 

```

[Installation log](https://pastebin.com/XmRefm34)

Prints of the rofi, respectively
![swappy-20240224-212449](https://github.com/Rolv-Apneseth/rofi-games/assets/102763838/bce02eef-ecb1-4864-a269-b040ee377757)

![swappy-20240224-212513](https://github.com/Rolv-Apneseth/rofi-games/assets/102763838/39b12671-4164-4536-997e-12bbc7d026c3)

**Comment by zDyanTB** [comment]
Ok, i'm a noob. I did share the prints of the base branch.


I reinstalled with the PR but the results are the same

**Comment by Rolv-Apneseth** [comment]
The error about the theme is just cause I'm assuming you don't have a `games` theme? The theme that comes with this is `games-default` and `games` is just what I would name a theme that builds on top of that one.

> /usr/lib64/rofi/

So this is where the `just install` command would try to put `games.so`. Ensure that the manual install you had at `/usr/lib/rofi` is removed, though I assume if you're running that `find` command with `sudo` then it should(?) get everything.

Just to check, you cloned the repo then switched to the branch [fix-rofi-version-check](https://github.com/Rolv-Apneseth/rofi-games/tree/fix-rofi-version-check) using `git checkout fix-rofi-version-check`? I see in the build step that it's still applying the flag.

I'm quite stumped though. I might try set up a VM if the problem persists, you're on Fedora with Wayland right?

**Comment by zDyanTB** [comment]
Yes, games.rasi is only a theme error.

I ran `rm` with `find` ,  the files are gone. I switched to the branch and the flag is gone as well.
[log](https://pastebin.com/GXKZ5UDV)
![image](https://github.com/Rolv-Apneseth/rofi-games/assets/102763838/92af4470-3703-4324-93b0-52a91efa2886)


Yes. It is Fedora 39. I've used this [script](https://github.com/JaKooLit/Fedora-Hyprland)

**Comment by Rolv-Apneseth** [comment]
So it launches now? Do you have games installed and it's not finding them?

**Comment by Rolv-Apneseth** [comment]
Sorry, just saw that error at the bottom of your logs. Seems like a separate issue so I'm opening one on the repo responsible for finding games and moving the rest of the discussion there. Thanks for the help with this one!
